src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallLibertyTask.groovy

```diff
@@ -38,11 +43,79 @@ class InstallLibertyTask extends AbstractTask {
         if (project.liberty.install.version != null) {
             result.put('version', project.liberty.install.version)
         }
-
-        if (project.liberty.install.runtimeUrl != null) {
-            result.put('runtimeUrl', project.liberty.install.runtimeUrl)
+        
+        if (project.liberty.install.type != null) {
+            result.put('type', project.liberty.install.type)
         }
+        
+        // Maven repository is higher precedence over Liberty repository 
+        if (project.liberty.assemblyArtifact.artifactId != null ||
+            project.liberty.assemblyArtifact.version != null) {
+            
+            String groupId = 'com.ibm.websphere.appserver.runtime'
+            String version 
+            String artifactId = 'wlp-webProfile7'
+            String type = 'zip'
 
+            if (project.liberty.assemblyArtifact.groupId != null) {
+                groupId = project.liberty.assemblyArtifact.groupId
+            }
+            if (project.liberty.assemblyArtifact.artifactId != null) {
+                artifactId = project.liberty.assemblyArtifact.artifactId
+            }
+
+            if (project.liberty.assemblyArtifact.version != null) {
+                version = project.liberty.assemblyArtifact.version
+            } else {
+                def maven_version = getLatestVersionFromMaven(artifactId)
+                if (maven_version != null) {
+                    version = maven_version
+                }
+            }
+            
+            if (project.liberty.assemblyArtifact.type != null) {
+                type = project.liberty.assemblyArtifact.type
+            }
+            
+            def ss = project.getConfigurations().getNames()
+            if (!ss.contains('InstallLibertyTaskConfig')) {
+                project.getConfigurations().create('InstallLibertyTaskConfig')
+                project.getDependencies().add('InstallLibertyTaskConfig', groupId + ':' +
```

Comment:
Correct. By adding dependency, it will make download the file into the local Gradle repository.